Updating `mcs.read()` in the example Lua files

YufeiHu opened the pull request:
Updating `mcs.read()` in the example Lua files to incorporate the API change for `mcslib_read()` introduced in this previous change: https://github.com/memcached/mcshredder/commit/011b36b372de086c1ccfa7d1b89e703d8b195fdd 

### Test 
I was able to run `mcshredder` using the updated `example.lua` (with key prefix set to some actual value). I did `watch proxyreqs` with sampling disabled and could see below from the proxy:
```
...
ts=1724284316.964780 gid=518230 type=proxy_req elapsed=755 type=101 code=15 status=0 cfd=0 be=10.100.88.89:11216 detail=fg_cache_hit req=mg /test/14 v
...
```

feihu-stripe reviewed (approved):
LGTM

dormando commented on conf/metafactory.lua:
```diff
@@ -56,7 +56,8 @@ function basic(a)
             mcs.write(set)
             mcs.flush()
-            local res = mcs.read()
+            local res = mcs.res_new()
```
this code is commented and I don't really remember why.

for note though: Since this is an 'init' style mcshredder function (returns a function) the res can/should be pre-allocated with the other one above, ie:

`local set_res = mcs.res_new()` then use that down in this commented code instead of creating one to throw away

YufeiHu replied:
Makes sense, updated.

dormando commented:
I think this is an okay change; it at least makes the examples work again. However the example is pretty out of date and I or someone should update it. The 'init' format allows pre-creating objects and there're API's for checking lines without allocating strings. I probably have tons of local config files that I never committed that do this... but also the shredders repo is full of these style functions.

dormando commented:
would you feel up to attempting to update the example file further? if not I'll do it when I get a chance.

YufeiHu commented:
> would you feel up to attempting to update the example file further? if not I'll do it when I get a chance.

Yea I will be happy to help with the update in a separate PR. Please let me know what specific updates you are looking for.

Also I tested running the `mcshredder` using `dumpload_testdata.lua`, `example.lua`, `loader.lua`, `metafactory.lua` on my devbox (with `prefix` changed to the actual value) to send traffic to a local proxy and then reroute to some testing hosts in QA, and they all seemed to work: I could see the requests arriving at the proxy by doing `watch proxyreqs` with sampling disabled.

dormando commented on conf/metafactory.lua:
```diff
@@ -56,7 +56,8 @@ function basic(a)
             mcs.write(set)
             mcs.flush()
-            local res = mcs.read()
+            local set_res = mcs.res_new()
```
think you misunderstood my request:

- Move `local set_res = res_new()` to where line 45 is.
- This allows reuse of the object on each execution run.

YufeiHu replied:
Oh right, because it's in a function that's being returned. 
Sorry about that! Please let me know if I need to make changes elsewhere, thanks!

dormando commented:
Looks good now, thanks!

The next step is you can squash the commits down and update the commit message. If you look at the commit history I try to be descriptive, ie:

```
70 character or less summary/intent
[empty line]
longer description if necessary
```

In this case instead of "Updating examples" you'd hint as to the actual contents: "Updating `mcs.read()` examples for `mcp.res_new()`

Unlike corpo repos the commit history of OSS projects is supposed to be both legible and bisectable.

YufeiHu commented:
> Looks good now, thanks!
> 
> The next step is you can squash the commits down and update the commit message. If you look at the commit history I try to be descriptive, ie:
> 
> ```
> 70 character or less summary/intent
> [empty line]
> longer description if necessary
> ```
> 
> In this case instead of "Updating examples" you'd hint as to the actual contents: "Updating `mcs.read()` examples for `mcp.res_new()`
> 
> Unlike corpo repos the commit history of OSS projects is supposed to be both legible and bisectable.

Thanks! I’ve just squashed all the commits into a single commit and updated the commit message to hopefully meet the requirement.

dormando commented:
merged, thanks!
